scripts/Packages/Package_Gullwing__QFP_SOIC_SO/Readme.md

```diff
@@ -0,0 +1,121 @@
+Examples of supported packages include: QFN, SOIC, SO
+
+## Running the script
+
+The script requires python 3.4 or newer. Run it with:
+`python3 ipc_gullwing_generator.py size_definitions/eqfp.yaml` (replace eqfp.yaml with the size definition file that contains your part.)
+
+## Size definition format
+
+Every file contains a header with parameters applied to all parts. These define the common footprint name prefix and the output library.
+
+``` yaml
+FileHeader:
+  library_Suffix: 'QFP' #resulting library name Package_QFP
+  device_type: 'EQFP' #footprint names will be of style ...EQFP-pincount_...
+```
+
+---
+
+Every further entry in the script is assumed to be a package size definition. The top level parameter will give the internal parameter set name. This one must be unique in this file and should be representative of the footprint as it will be used in error messages. (It also makes later maintenance easier if it is easy to determine which footprint was genererated by which parameter set.)
+
+``` yaml
+internal_package_name:
+  # parameter list (See below)
+```
+
+### Documentation and naming parameters
+- Size source to be added to footprint documentation field (`size_source`) {url}
+- footprint name control
+  - Manufacturer if specific to one manufacturer (`manufacturer`) {string}
+  - Part number if specific to one part (`part_number`) {string}
+  - Custom naming (`custom_name_format`) {format string}
+
+### Package dimensions
+![dimension example](./documentation/dimension_system.svg)
+- Body size (`body_size_x`, `body_size_y`, `overall_height`) {dimension}
+- Lead dimensions:
+  - Over all size representing lead tip to lead tip dimension (`overall_size_x`, `overall_size_y`) {dimension}
+  - Lead with (`lead_width`) {dimension}
+  - Lead length representing tip till bend (`lead_len`)
+- lead pitch, currently equal for all sides (`pitch`) {float}
+- pin count, 0 means no pins in that direction (`num_pins_x`, `num_pins_y`) {int}
+
+
+### Exposed pad handling:
+![exposed pad example](../documentation/ep_handling.svg)
+- Size of package exposed pad or slug (`EP_size_x`, `EP_size_y`) {dimension}
+- Optional the size of the footprint pad (`EP_size_x_overwrite`, `EP_size_y_overwrite`) {float}
+   - Pad size is equal to nominal package pad size if not given.
+- Optional the size of the mask cutout (`EP_mask_x`, `EP_mask_y`) {float}
+   - Use to create solder mask defined pads (in combination with `EP_size_x_overwrite`)
+- Paste is typically split into a regular grid with (`EP_num_paste_pads`) {[int (x), int (y)]}
+  - The paste coverage multiplicator determines how much of the exposed copper area is covered by paste. (`EP_paste_coverage`) {float (0..1), default=0.65}
+
+### Thermal vias
+A package with exposed pad can generate a version with thermal vias. This will be generated in addition to the normal version.
+
+![exposed pad example](../documentation/thermal_vias.svg)
```

Comment:
The image adds a lot of clarify but I have a few questions:
1. The image doesn't show the meaning of each item by color. Blue is copper. Pink is paste. Is the orange-ish color mask?
1. In the first row, what is creating the orange-ish round rect? Is it mask as I guessed above? What setting?
1. Similarly, the bottom two images show different padstack but the info below them in the same.

Reply:
Clarified image